src/main/java/org/broadinstitute/dsm/pubsub/WorkflowStatusUpdate.java

```diff
@@ -32,11 +39,48 @@ public static void updateCustomWorkflow(Map<String, String> attributesMap, Strin
 
         List<ParticipantDataDto> participantDatas = participantDataDao.getParticipantDataByParticipantId(ddpParticipantId);
 
-        participantDatas.forEach(participantDataDto -> {
-            updateProbandStatusInDB(workflow, status, participantDataDto, studyGuid);
+        if (!participantDatas.isEmpty()) {
+            participantDatas.forEach(participantDataDto -> {
+                updateProbandStatusInDB(workflow, status, participantDataDto, studyGuid);
+            });
+        } else {
+            addNewParticipantDataWithStatus(workflow, status, ddpParticipantId);
+        }
+
+        exportToESifNecessary(workflow, status, ddpParticipantId, instance);
+    }
+
+    public static void exportToESifNecessary(String workflow, String status, String ddpParticipantId, DDPInstance instance) {
+        Optional<FieldSettingsDto> fieldSetting = fieldSettingsDao.getFieldSettingByColumnName(workflow);
+        fieldSetting.ifPresent(setting -> {
+            String actions = setting.getActions();
+            if (actions != null) {
+                Value[] actionsArray =  gson.fromJson(actions, Value[].class);
+                for (Value action : actionsArray) {
+                    if (ESObjectConstants.ELASTIC_EXPORT_WORKFLOWS.equals(action.getType())) {
+                        ElasticSearchUtil.writeWorkflow(instance, ddpParticipantId, workflow, status);
+                        break;
+                    }
+                }
+            }
         });
+    }
 
-        ElasticSearchUtil.writeWorkflow(instance, ddpParticipantId, workflow, status);
+    public static int addNewParticipantDataWithStatus(String workflow, String status, String ddpParticipantId) {
+        Optional<FieldSettingsDto> fieldSetting = fieldSettingsDao.getFieldSettingByColumnName(workflow);
+        JsonObject dataJsonObject = new JsonObject();
+        dataJsonObject.addProperty(workflow, status);
+        AtomicInteger participantDataId = new AtomicInteger();
```

Comment:
If you wanted to avoid using of AtomicInteger you could use `map` of optional, like:
fieldSetting  
    \.map(setting -> {//dao/dto stuff here})
    \.orElse(0)